src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp

```diff
@@ -768,13 +769,19 @@ RangesInDataParts MergeTreeDataSelectExecutor::filterPartsByPrimaryKeyAndSkipInd
                 parts_with_ranges[part_index] = std::move(ranges);
         };
 
-        size_t num_threads = std::min<size_t>(num_streams, parts.size());
+        size_t num_threads = parts.size();
         if (settings[Setting::max_threads_for_indexes])
         {
-            num_threads = std::min<size_t>(num_streams, settings[Setting::max_threads_for_indexes]);
```

Comment:
imo it is ok to not lower it to `num_streams`, but maybe we should still be bounded by `max_threads`?

Reply:
My concern here is that previously it was always 1 i.e. single-threaded and if we limit it by `max_threads` it will change the default behavior and potentially introduce regressions.

Reply:
Ok, it's not always disabled, it's just for query with 'LIMIT 1'.  Then I'll change the bound to `max_threads` as you suggest